### ISSUE TITLE
3.0.0 - Errors when running typedoc on code that uses date-fns

### ISSUE THREAD

**Issue by nstuyvesant** [date-fns/date-fns#3616]
Working on migrating `@carbon/charts` to `date-fns@3.0.0`.  When we produce documentation using typedoc, getting dozens of errors like:
```
src/components/essentials/tooltip.ts:184:11 - error TS1362: 'format' cannot be used as a value because it was exported using 'export type'.

184    return format(value, 'MMM d, yyyy')
              ~~~~~~

  ../../node_modules/date-fns/index.d.mts:1:1
    1 export type * from "./index.d.ts";
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'format' was exported here.
```

If I do a global replace in node_modules/date-fns/package.json of ".mts" with ".ts", the problem is resolved (locally).

Not sure the .d.mts files are really necessary especially since they just re-export everything in the .d.ts files anyway.

**Comment by b-maslennikov** [comment]
Issue #3588

**Comment by kossnocorp** [comment]
Can someone please produce a minimal reproduction repo? It will help to fix the problem.

We have a set of different environments [where it's all good](https://github.com/date-fns/date-fns/actions/runs/7247215749/job/19740705115): https://github.com/date-fns/date-fns/tree/main/examples

Sorry that it slipped!

**Comment by kossnocorp** [comment]
I shipped with fix with `date-fns@3.0.1`, please try it

**Comment by nstuyvesant** [comment]
Appears to have been resolved - thanks.
